src/components/elements/button/index.tsx

```diff
@@ -0,0 +1,33 @@
+import { StyledButton, StyledButtonWrapper } from './button.styles';
+import type { ButtonProps } from './button.types';
+import { MovingBorder } from '../moving-border';
+
+export const Button = async ({
+  variant = 'primary',
+  small = false,
+  type = 'button',
+  withBorder = false,
+  children,
+  as = 'button',
+  icon: Icon,
+  iconWeight = 'fill',
+  onClick,
+  href,
+}: Component<ButtonProps>) => {
+  const base = () => (
```

Comment:
Can we render it as JSX to make it more readable?
```suggestion
  const BaseButton = () => (
```
Then...

``` jsx
...
<BaseButton />
...
```